CallAutomation_OutboundCalling/src/main/java/com/communication/callautomation/ProgramSample.java

```diff
@@ -0,0 +1,194 @@
+import com.azure.communication.callautomation.models.*;
+import com.azure.communication.callautomation.models.events.*;
+import com.azure.communication.common.CommunicationIdentifier;
+import com.azure.communication.common.PhoneNumberIdentifier;
+import com.azure.communication.identity.implementation.models.CommunicationErrorResponseException;
+import com.azure.core.http.rest.Response;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+
+@RestController
+@Slf4j
+public class ProgramSample {
+    private AppConfig appConfig;
+    private CallAutomationAsyncClient asyncClient;
+
+    public ProgramSample(final AppConfig appConfig) {
+        this.appConfig = appConfig;
+        asyncClient = initClient();
+    }
+
+    @GetMapping(path = "/api/outbound_call")
+    public ResponseEntity<String> handleOutboundCall() {
+        String callConnectionId = createOutboundCall();
+        return ResponseEntity.ok().body("Target participant: "
+                + appConfig.getTargetphonenumber() +
+                ", CallConnectionId: " + callConnectionId);
+    }
+
+    @PostMapping(path = "/api/callback")
+    public ResponseEntity<String> handleCallbackEvents(@RequestBody final String reqBody) {
+        List<CallAutomationEventBase> events = CallAutomationEventParser.parseEvents(reqBody);
+        for (CallAutomationEventBase event : events) {
+            String callConnectionId = event.getCallConnectionId();
+            if (event instanceof CallConnected) {
+                //start recording
+                String callRecordingId = callRecording(callConnectionId);
+                log.info("Call recording started with ID: {}", callRecordingId);
+                // recognize tones
+                String statusCode = recognizeDtmfTones(callConnectionId, "MainMenu.wav");
+                log.info("DTMF recognition started. Request status code {}", statusCode);
+            }
+            else if (event instanceof RecognizeCompleted) {
+                log.info("Recognize Completed event received");
+                RecognizeCompleted recognizeEvent = (RecognizeCompleted) event;
+                DtmfResult dtmfResult = (DtmfResult) recognizeEvent
+                        .getRecognizeResult().get();
+                DtmfTone selectedTone = dtmfResult.getTones().get(0);
+
+                switch(selectedTone.convertToString()) {
+                    case "1":
+                        log.info("Received DTMF tone 1.");
+                        log.info("Played media file with {}", playToAll(callConnectionId, "Confirmed.wav"));
```

Comment:
Can we call method directly instead of calling in log APIs. I feel it affects code readability.

Reply:
applied suggestion, thanks